lib/tasks/schools_data.rake

```diff
@@ -50,4 +51,48 @@ namespace :schools_data do
 
     puts "Done! created: #{created}, updated: #{updated}"
   end
+
+  desc "Generate data/lead_schools_publish.csv from Publish api"
+  task :generate_csv_from_publish, %i[output_path] => [:environment] do |_, args|
+    request_uri = TeacherTrainingApi::RetrieveLeadSchools::DEFAULT_PATH
+
+    items = []
+
+    while request_uri
+      payload = TeacherTrainingApi::RetrieveLeadSchools.call(request_uri:)
+      items += payload[:data].map do |provider_data|
+        provider_data[:attributes].slice(:urn, :name, :city, :postcode)
+      end
+
+      request_uri = payload[:links][:next]
+    end
+
+    items = items.uniq { |a| a[:urn] }
+    items = items.sort { |a, b| a[:urn] <=> b[:urn] }
+
+    CSV.open(args.output_path || PUBLISH_CSV_PATH, "w+") do |csv|
+      csv << items.first.keys
+      items.each do |hash|
+        csv << hash.values
+      end
+    end
+  end
+
+  desc "Update schools to lead schools from csv data/lead_schools_publish.csv"
+  task :update_school_to_lead_school_publish, %i[input_path] => [:environment] do |_, args|
```

Comment:
could we just rename this to `update_lead_schools_from_publish`?

Reply:
Done